### PR TITLE
Android Sample: update Gradle

### PR DESCRIPTION
**Pull request by devemux86**
- Update Gradle plugin 2.1.0
- Include `gradle.properties` for [dexing-in-process](http://developer.android.com/tools/revisions/gradle-plugin.html#revisions)

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# any settings specified in this file.
+
+# For more details on how to configure your build environment visit
+# http://www.gradle.org/docs/current/userguide/build_environment.html
+
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+# Default value: -Xmx10248m -XX:MaxPermSize=256m
+# org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+org.gradle.jvmargs=-Xmx2048m
```
Comment (karussell) [inline android/gradle.properties]: oh, wow. Is that really necessary so high?

Comment (devemux86) [inline android/gradle.properties]: Yes I'm afraid it's the official instruction for enabling dexing-in-process, see [here](http://developer.android.com/tools/revisions/gradle-plugin.html#revisions) (I updated the PR doc).

Comment (devemux86) [inline android/gradle.properties]: It's just an option for faster builds.
If you feel uncomfortable with that we can remove it.

Comment (karussell) [inline android/gradle.properties]: I think this would just fail in my personal case having open other apps as well :)

**Pull request merged**